### ISSUE TITLE
Add aria-label to Blacklight::Icon

### ISSUE THREAD

**Issue by mejackreed** [projectblacklight/blacklight#2300]
Previously we used `aria-labelledby` but the duplicate ids caused invalid HTML. Removed in https://github.com/projectblacklight/blacklight/pull/2157 This proposal is to add `aria-label` in its place now.
